engine/src/main/java/org/terasology/engine/world/block/DefaultColorSource.java

```diff
@@ -17,29 +17,23 @@ public Colorc calcColor(int x, int y, int z) {
     COLOR_LUT {
         @Override
         public Colorc calcColor(int x, int y, int z) {
+            ColorProvider colorProvider = CoreRegistry.get(ColorProvider.class);
```

Comment:
Is doing this lookup in `CoreRegistry` fast? I think @tolziplohu put this optimization on purpose to reduce the necessary lookups in the registry with each lookup of a color... :thinking:

Reply:
yea, but we can't save it as a static variable.